lntest/itest/lnd_test.go

```diff
@@ -10145,7 +10145,7 @@ func testBidirectionalAsyncPayments(net *lntest.NetworkHarness, t *harnessTest)
 	}
 
 	// Calculate the number of invoices.
-	numInvoices := int(info.LocalBalance / paymentAmt)
+	numInvoices := input.MaxHTLCNumber / 2
```

Comment:
Do we know why this didn't fail before?

Reply:
Since the new check is on the receive side that could mean we are not properly checking sender side we are not overflowing the commitment.

Reply:
You're right - the last commit is not the correct fix.  The sender is overflowing the commitment, but there is a check there. I suspect an off-by-one error somewhere...